### PR TITLE
fix(pin-input): disable input when isDisabled or isReadonly is true

### LINKED ISSUE #3404: `PinInputField` Allows Input When `isDisabled` or `isReadOnly ` is Set
### Which package has an issue?

@yamada-ui/react

### Package Version

1.6.2

### Description

Bug that allows input even if `isDisabled` or `isReadOnly` is set to PinInputField.

It is possible to reproduce the following.
https://yamada-ui.github.io/yamada-ui/?path=/story/components-forms-pininput--is-disabled
https://yamada-ui.github.io/yamada-ui/?path=/story/components-forms-pininput--is-readonly

### Link to Reproduction

_No response_

### Steps to Reproduce

2. Click on '...'
3. Scroll down to '...'

### Browser

Chrome

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iPadOS
- [ ] iOS
- [ ] Android

### Do you know how to fix the issue?

None

### Are you willing to participate in fixing this issue and create a pull request with the fix?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by miyamyia**
Check disabled and readonly status for both PinInput and PinInputField

Closes #3404 

## Description

Fix disabled/readonly state handling for both PinInput and PinInputField components to ensure proper input restrictions.

## Current behavior (updates)

Previously, PinInputField only checked the disabled and readonly status of its parent PinInput component, ignoring its own isDisabled and isReadOnly props. This meant that even if a PinInputField was individually set as disabled or readonly, it would only respect the parent PinInput's state.

## New behavior

- PinInputField now checks both its own disabled/readonly status and its parent PinInput's state
- Input is properly disabled when either PinInputField's isDisabled/isReadOnly is true or when parent PinInput's props are set
- This provides more granular control over individual PinInputFields while maintaining parent-level control

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Review by nathanlao — DISMISSED** [review]
Thank you for the contribution, I don't see any problems in this fix :)

File: packages/components/pin-input/src/pin-input.tsx [hunk 1]
```diff
@@ -335,12 +335,12 @@ export const PinInput = forwardRef<PinInputProps, "div">(
         ...filterUndefined(props),
         id: `${id}-${index}`,
         autoComplete: otp ? "one-time-code" : "off",
-        disabled,
+        disabled: props.disabled || disabled,
```
Comment (hirotomoyamada) [inline packages/components/pin-input/src/pin-input.tsx:338]: Remove this line.

File: packages/components/pin-input/src/pin-input.tsx [hunk 2]
```diff
@@ -335,12 +335,12 @@ export const PinInput = forwardRef<PinInputProps, "div">(
...
         placeholder:
           focusedIndex === index && !readOnly && !props.readOnly
             ? ""
             : placeholder,
-        readOnly,
+        readOnly: props.readOnly || readOnly,
```
Comment (hirotomoyamada) [inline packages/components/pin-input/src/pin-input.tsx:343]: Remove this line.

File: packages/components/pin-input/src/pin-input.tsx [hunk 3]
```diff
@@ -335,12 +335,12 @@ export const PinInput = forwardRef<PinInputProps, "div">(
         ...filterUndefined(props),
         id: `${id}-${index}`,
         autoComplete: otp ? "one-time-code" : "off",
-        disabled,
+        disabled: props.disabled || disabled,
         placeholder:
           focusedIndex === index && !readOnly && !props.readOnly
             ? ""
             : placeholder,
-        readOnly,
```
Comment (hirotomoyamada) [inline packages/components/pin-input/src/pin-input.tsx:343]: Move this line above `..filterUndefined(props)`.

File: packages/components/pin-input/src/pin-input.tsx [hunk 4]
```diff
@@ -335,12 +335,12 @@ export const PinInput = forwardRef<PinInputProps, "div">(
         ...filterUndefined(props),
         id: `${id}-${index}`,
         autoComplete: otp ? "one-time-code" : "off",
-        disabled,
```
Comment (hirotomoyamada) [inline packages/components/pin-input/src/pin-input.tsx:338]: Move this line above `..filterUndefined(props)`.

**Comment by miyamyia** [comment]
Thank you for your review. I overlooked the behavior of filterUndefined.

**Review by hirotomoyamada — APPROVED** [review]
Thanks!

**Pull request merged**